test/policies/test_result_policy_test.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class TestResultPolicyTest < ActiveSupport::TestCase
+  setup do
+    assert Pundit.policy_scope(users(:test), TestResult), []
+    users(:test).account = accounts(:test)
+    users(:test).save
+  end
+
+  test 'only test results within visible profiles are accessible' do
+    profiles(:one).account = accounts(:test)
+    profiles(:one).save
+    assert_includes Pundit.policy_scope(users(:test), Profile), profiles(:one)
+    test_results(:one).profile = profiles(:one)
+    rule_results(:one).host = hosts(:one)
+    rule_results(:one).rule = rules(:one)
+    test_results(:one).rule_results = [rule_results(:one)]
+    # Save without validations to force a TestResult with a profile
+    # but without a host
+    test_results(:one).save(validate: false)
```

Comment:
I had to do this here and in the other test to ensure we could test the 2nd code path where a TestResult has a Profile but no Host - this shouldn't be possible according to validations though.